cli/src/cli_builder.rs

```diff
@@ -325,11 +312,24 @@ impl<'a> NewCli<'a> {
                 break;
             }
         }
-        node.cmd
-            .as_ref()
-            .expect("no cmd for node")
-            .run_cmd(sm, &ctx)
-            .await
+        let cmd = node.cmd.as_ref().expect("no cmd for node");
+
+        if cmd.maybe_long() {
+            // Keep a reasonable timeout for initial connection.
+            client_config = client_config.with_connect_timeout(15);
+
+            // Set a long timeout for "maybe long" commands.
+            if timeout.is_none() {
+                client_config = client_config.with_timeout(60 * 60);
+            }
+
+            // Kill the connection if we stop receiving data before the
+            // connection timeout.
+            client_config = client_config.with_read_timeout(30);
+        }
+
+        let ctx = Context::new(client_config)?;
+        cmd.run_cmd(sm, &ctx).await
```

Comment:
@wfchandler would particularly like your eyes on this section

Reply:
This makes sense to me 👍